libethereum/EthereumHost.cpp

```diff
@@ -395,6 +412,8 @@ EthereumHost::EthereumHost(BlockChain const& _ch, OverlayDB const& _db, Transact
 EthereumHost::~EthereumHost()
 {
 	terminate();
+	m_peerObserver->disableSync(); // m_peerObserver contains a reference to BlockchainSync, so that should be disabled first.
```

Comment:
Can we instead just do `m_peerObserver.reset()` ?

Reply:
Ah I see, probably not, because peers still have shared_ptr to observer.

But then - can we make `BlockChainSync` member of `EthereumPeerObserver` to be shared_ptr ? This should solve the problem

Reply:
I'll try that.

Reply:
Done.